src/sage/schemes/elliptic_curves/kummer_line.py

```diff
@@ -0,0 +1,1765 @@
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: K = KummerLine(E); K.base_ring()
+            Finite Field of size 101
+        """
+        return self._base_ring
+
+    def zero(self):
+        r"""
+        Return the identity point on the Kummer Line.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: K = KummerLine(E); K.zero()
+            (1 : 0)
+        """
+        return self(None)
+
+    def curve(self):
+        r"""
+        Return the associated elliptic curve.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: K = KummerLine(E); K.curve()
+            Elliptic Curve defined by y^2 = x^3 + 2*x + 3 over Finite Field of size 101
+        """
+        return self._curve
+
+    @cached_method
+    def j_invariant(self):
+        r"""
+        Compute the j-invariant of the associated elliptic curve.
+
+        Cached method.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(j=42)
+            sage: K = KummerLine(E)
+            sage: K.j_invariant() == 42
+            True
+
+        ::
+
+            sage: K = KummerLine(GF(101), [2, 3])
+            sage: K.j_invariant()
+            74
+        """
+        return self._curve.j_invariant()
+
+    @cached_method
+    def discriminant(self):
+        r"""
+        Compute the discriminant of the associated elliptic curve.
+
+        Cached method.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: K = KummerLine(E); K.discriminant()
+            44
+        """
+        return self._curve.discriminant()
+
+    def isogeny(self, xP):
+        r"""
+        Compute the isogeny with kernel ``xP``.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3]); K = KummerLine(E)
+            sage: P = E(76, 65); xP = K(P)
+            sage: Q = E(49, 61); xQ = K(Q)
+            sage: KummerLineIsogeny(xP)
+            Isogeny of degree 12 from Kummer Line of Elliptic Curve defined by y^2 = x^3 + 2*x + 3 over Finite Field of size 101 to Kummer Line of Elliptic Curve defined by y^2 = x^3 + 23*x + 73 over Finite Field of size 101
+        """
+
+        if xP._parent != self:
+            raise ValueError("point is not on the Kummer line")
+        return KummerLineIsogeny(xP)
+
+
+class KummerLine_weierstrass(KummerLine_generic):
+    r"""
+    Kummer line class for a short Weierstrass elliptic curve.
+
+    EXAMPLES::
+
+        sage: E = EllipticCurve(GF(101), [2, 3])
+        sage: K = KummerLine(E); K
+        Kummer Line of Elliptic Curve defined by y^2 = x^3 + 2*x + 3 over Finite Field of size 101
+    """
+
+    def __init__(self, curve):
+        r"""
+        Constructor for a Kummer line from a short Weierstrass elliptic curve.
+
+        INPUT: ``curve`` -- an elliptic curve in short Weierstrass form `y^2 = x^3 + a*x + b`
```

Comment:
```suggestion
        INPUT: ``curve`` -- an elliptic curve in short Weierstrass form `y^2 = x^3 + ax + b`
```
Single quotations means the content inside is $\LaTeX$.